test/sampling.jl

```diff
@@ -5,6 +5,26 @@ import StatsBase: norepeat, randi
 
 srand(1234)
 
+# test that rng specification is working correctly
+# a) if the same rng is passed to a sample function twice, 
+#    the results should be the same (repeatability)
+# b) not specifying a rng should be the same as specifying Base.GLOBAL_RNG
+function test_rng_use(func, non_rng_args...)
+    # some sampling methods mutate a passed array and return it
+    # so that the tests don't pass trivially, we need to copy those
+    # pre-allocated storage arrays
+
+    # repeatability
+    @test func(MersenneTwister(1), deepcopy(non_rng_args)...) ==
```

Comment:
I don't think `copy` is enough. Suppose `func` is one of the sampling methods that writes into a passed array. If just `copy` is used, both the left hand side and right hand side `func` calls will be writing into the same array. So the equality test just ends up testing that an array is equal to itself. For example:

```
using Base.Test

function test_rng_use(func, non_rng_args...)
    # repeatability
    @test func(MersenneTwister(1), copy(non_rng_args)...) ==
          func(MersenneTwister(1), copy(non_rng_args)...)
end

# should fail
function ignores_rng!(r::AbstractRNG, x::Vector{Float64})
	for i in eachindex(x)
		x[i] = rand()
	end
	return x
end
ignores_rng!(x) = ignores_rng!(Base.GLOBAL_RNG, x)

test_rng_use(ignores_rng!, [1.0, 2.0])
```
run in Julia:
```
julia> include("C:\\Users\\Evan\\Desktop\\randtest.jl")
WARNING: copy(x::Tuple) is deprecated, use identity(x) instead.
 in depwarn(::String, ::Symbol) at .\deprecated.jl:64
 in copy(::Tuple{Array{Float64,1}}) at .\deprecated.jl:50
 in test_rng_use(::Function, ::Array{Float64,1}, ::Vararg{Array{Float64,1},N}) at C:\Users\Evan\Desktop\randtest.jl:5
 in include_from_node1(::String) at .\loading.jl:488
 in eval(::Module, ::Any) at .\boot.jl:234
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at .\REPL.jl:64
 in macro expansion at .\REPL.jl:95 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at .\event.jl:68
while loading C:\Users\Evan\Desktop\randtest.jl, in expression starting on line 18
Test Passed
  Expression: func(MersenneTwister(1),copy(non_rng_args)...) == func(MersenneTwister(1),copy(non_rng_args)...)
   Evaluated: [0.367674,0.409404] == [0.367674,0.409404]

julia>
```